src/main/java/com/xpanxion/java/springboot/da1/demo/controller/student9/MemberController9.java

```diff
@@ -32,4 +30,9 @@ public MemberHistory9 addMemberCheckOutHistory(@PathVariable int memberId, @Requ
 
         return memberHistoryService9.addMemberCheckOutHistory(memberId, time);
     }
+
+    @GetMapping("student9/api/v1/member/{memberId}/workout-history")
+    public List<MemberHistory9> getMemberHistory(@PathVariable int memberId){
+        return memberHistoryService9.getMemberWorkOutHistory(memberId);
```

Comment:
Can we try to match the requirements for this assignment as best as possible?  I'm looking for this format:

```
[
    {
        "memberId": 32,
        "timeUtc": "2022-01-02T19:48:28.000+00:00",
        "checkType": "CHECK_IN"
    },
    {
        "memberId": 32,
        "timeUtc": "2022-01-02T19:48:28.000+00:00",
        "checkType": "CHECK_OUT"
    },
    {
        "memberId": 32,
        "timeUtc": "2022-01-02T14:48:28.000+00:00",
        "checkType": "CHECK_IN"
    },
    {
        "memberId": 32,
        "timeUtc": "2022-01-02T15:48:28.000+00:00",
        "checkType": "CHECK_OUT"
    }
]
```

Reply:
{
Member Id: 7
CheckIn: 2021-12-12T09:00
CheckOut: null
}

{
Member Id: 7
CheckIn: null
CheckOut: 2021-12-12T09:35
}

{
Member Id: 7
CheckIn: null
CheckOut: 2021-12-14T10:35
}

{
Member Id: 7
CheckIn: 2021-12-14T10:35
CheckOut: null
}

So i couldnt figure out a way just by building a list but i was able to get this using a Stringbuilder. Is that ok?

Reply:
The `StringBuilder` approach is ok for now, but you should practice streams and the map function (`stream().map()`).  Just fire up a new console app and nerd out with streams.  I have some examples in my reference repos.  https://youtu.be/CDMVaQOvtxU